cmd/consensus/main.go

```diff
@@ -616,10 +611,12 @@ func main() {
 				mainMetrics,
 			)
 
+			telemetryConsumer := notifications.NewTelemetryConsumer(logger, notifier)
+
+			// TODO(leo): move these to NewTelemetryConsumer
 			notifier.AddParticipantConsumer(telemetryConsumer)
 			notifier.AddCommunicatorConsumer(telemetryConsumer)
-			notifier.AddFinalizationConsumer(telemetryConsumer)
-			notifier.AddFollowerConsumer(followerDistributor)
+			notifier.ProposalViolationDistributor.AddProposalViolationConsumer(followerDistributor)
```

Comment:
We removed `notifier.AddFollowerConsumer(followerDistributor)`, because it's subscribing both OnBlockFinalized and OnBlockIncorporated events, which is redundant. 

The only additional thing it does is `AddProposalViolationConsumer`, so we added it here. 

Future TODO: this can also be moved to the `NewTelemetryConsumer`